transport/src/main/java/io/netty/channel/ChannelId.java

```diff
@@ -43,7 +43,7 @@
  * </ul>
  * </p>
  */
-public interface ChannelId extends Serializable, Comparable<ChannelId> {
+public interface ChannelId<T extends ChannelId> extends Serializable, Comparable<T> {
```

Comment:
this is an API change. I don't think we can do this in a minor release. Can you revert this change please?